testutils/src/test/resources/xml_examples/example12_diff.txt

```diff
@@ -0,0 +1,10 @@
++-----------------------+-------------+-----------+------------+-------------+--------------+---------------------+-----------+---------+----------+-----------+------------+------------------------------------------------------------+
+|expected__schemaVersion|expected_ckey|expected_id|expected_id2|expected_name|expected_value|actual__schemaVersion|actual_ckey|actual_id|actual_id2|actual_name|actual_value|errCol                                                      |
++-----------------------+-------------+-----------+------------+-------------+--------------+---------------------+-----------+---------+----------+-----------+------------+------------------------------------------------------------+
+|null                   |null         |null       |null        |null         |null          |1.0.0                |[a, b10]   |3        |1         |GGG        |812.0       |[value, name, id2, id, ckey_value, ckey_key, _schemaVersion]|
+|1.0.0                  |[a, b1]      |1          |1           |AAA          |77.222        |1.0.0                |[a, b1]    |1        |1         |AAA        |77.1        |[value]                                                     |
+|1.0.0                  |[a, b2]      |1          |2           |BBB          |11.2          |1.0.0                |[a, b2]    |1        |2         |BBB        |11.1        |[value]                                                     |
+|1.0.0                  |[a, b4]      |2          |1           |DDD          |1000000.1     |1.0.0                |[a, b4]    |2        |1         |DDD1       |1000000.1   |[name]                                                      |
+|1.0.0                  |[a, b5]      |2          |2           |EEE          |-9123.0       |1.0.0                |[a, b5]    |2        |2         |EEE2       |-9123.0     |[name]                                                      |
+|1.0.0                  |[a, b9]      |3          |1           |GGG          |812.0         |null                 |null       |null     |null      |null       |null        |[value, name, id2, id, ckey_value, ckey_key, _schemaVersion]|
++-----------------------+-------------+-----------+------------+-------------+--------------+---------------------+-----------+---------+----------+-----------+------------+------------------------------------------------------------+
```

Comment:
Just wondering... would it be safer to use Datasets/ case classes for the actual comparison?

This seems like it could be fragile should the format of this change in diff versions of spark etc.

Reply:
Yeah, it is fragile to `.show()` formatting. Case classes might be better. But now that you've mentioned it, I'm thinking about using conversion to JSON for comparison. It is format independent and the code won't be bloated with additional case classes. What do you think?

Reply:
Definitely better than show :)

Strongly typed way would probably be my preference, but json is fine by me, will be easier to read diff.